src/enumo/ruleset.rs

```diff
@@ -314,35 +298,53 @@ impl<L: SynthLanguage> Ruleset<L> {
         file.write_all("]".as_bytes()).expect("write failed");
     }
 
-    pub fn baseline_compare_to(
+    pub fn write_derivability_results(
         &self,
-        baseline: &Self,
-        baseline_name: &str,
-        domain_name: &str,
-        duration: Duration,
+        derive_type: DeriveType,
+        baseline: Self,
+        name: &str,
         limits: Limits,
-    ) {
-        self.write_json_equiderivability(
-            DeriveType::Lhs,
-            baseline,
-            &format!("{}_{}_lhs.json", baseline_name, domain_name),
-            limits,
-            duration,
-        );
-        self.write_json_equiderivability(
-            DeriveType::LhsAndRhs,
-            baseline,
-            &format!("{}_{}_lhs_rhs.json", baseline_name, domain_name),
-            limits,
-            duration,
-        );
+    ) -> ((String, String), (Duration, Duration)) {
+        let mut filepath = "rep/json/derivable_rules/".to_owned();
+
+        std::fs::create_dir_all(filepath.clone())
+            .unwrap_or_else(|e| panic!("Error creating dir: {}", e));
+
+        filepath.push_str(name);
+
+        match derive_type {
+            DeriveType::Lhs => filepath.push_str("_lhs.json"),
+            DeriveType::LhsAndRhs => filepath.push_str("_lhs_rhs.json"),
+            _ => filepath.push_str("_all.json"),
+        }
+
+        let mut file = std::fs::File::create(filepath.clone())
+            .unwrap_or_else(|_| panic!("Failed to open '{}'", filepath.clone()));
+
+        let start_f = Instant::now();
+        let (can_f, cannot_f) = self.derive(derive_type, &baseline, limits);
+        let time_f = start_f.elapsed();
+        let start_b = Instant::now();
+        let (can_b, cannot_b) = baseline.derive(derive_type, self, limits);
+        let time_b = start_b.elapsed();
+
+        let derivability_results = json!({
+            "forwards derivable": &can_f.to_str_vec(),
```

Comment:
Don't use "forwards" and "backwards" anywhere. As we've talked about, that wording can be confusing so it's better to say `enumo_derives_baseline` or `baseline_erives_enumo`